tests/app/oidc/test_gcp_oidc.py

```diff
@@ -0,0 +1,65 @@
+import time
+
+from cachetools.func import ttl_cache
+from freezegun import freeze_time
+from mock import Mock, patch
+
+from app.oidc.gcp_oidc import TTL, OIDCCredentialsServiceGCP
+
+TEST_SDS_OAUTH2_CLIENT_ID = "TEST_SDS_OAUTH2_CLIENT_ID"
+
+
+@patch("app.oidc.gcp_oidc.Request")
+@patch("app.oidc.gcp_oidc.fetch_id_token_credentials")
+def test_get_credentials(mock_token_fetch, mock_request):
```

Comment:
One for thought, it might be good to have an example of the previous cache test one you had for the purpose of if an expection is raised then it is not cached. It might be a given. So happy to leave depending on your thoughts.

Reply:
Happy with that - added a simple one that checks the cache stays empty if it fails